core/src/main/java/com/uber/m3/tally/ScopeImpl.java

```diff
@@ -49,7 +49,7 @@
     private final ConcurrentHashMap<String, GaugeImpl> gauges = new ConcurrentHashMap<>();
     private final ConcurrentHashMap<String, HistogramImpl> histograms = new ConcurrentHashMap<>();
 
-    private final ConcurrentLinkedQueue<Reportable> reportingQueue = new ConcurrentLinkedQueue<>();
+    private final CopyOnWriteArrayList<Reportable> reportingList = new CopyOnWriteArrayList<>();
```

Comment:
good call to switch here to CopyOnWriteArrayList. Usually number of reporters should remain the same